### PR TITLE
fix: added warning message for hydrator mutator

### PR DESCRIPTION
**Pull request by gen1us2k**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: docs/oathkeeper/pipeline/mutator.md
Comment (tomekpapiernik) [inline docs/oathkeeper/pipeline/mutator.md:482]: Hey man, I have a few problems with this warning. I think I know what you mean but I'm not sure.

- Oathkeeper needs to get custom headers with names that follow this format `UPPERCASE-UPPERCASElowercase`right? 
- Is there such a thing as "canonical HTTP headers" that defines the format ? When I google that I get articles about `rel=”canonical”` headers.

Comment (gen1us2k) [reply docs/oathkeeper/pipeline/mutator.md:482]: - UPPERCASE-UPPERCASElowercase-UPPERCASE-lowercase
- This is Go specific thing and related to the Go programming language. [Header](https://pkg.go.dev/net/http#Header) documentation and [textproto](https://pkg.go.dev/net/textproto#CanonicalMIMEHeaderKey) documentation

Comment (tomekpapiernik) [reply docs/oathkeeper/pipeline/mutator.md:482]: Do all custom headers have to start with `X-`? 

So the rule is `X-[Uppercase]lowercase-[Uppercase]lowercase(...)`?

Comment (gen1us2k) [reply docs/oathkeeper/pipeline/mutator.md:482]: >Do all custom headers have to start with X-?
Basically yes

>So the rule is X-[Uppercase]lowercase-[Uppercase]lowercase(...)?
Yep.

**Comment by gen1us2k** [comment]
BTW. The fix was merged https://github.com/ory/oathkeeper/pull/951 hence the warning needs to be updated

File: docs/oathkeeper/pipeline/mutator.md
Comment (tomekpapiernik) [inline docs/oathkeeper/pipeline/mutator.md:485]: Here's how I'd write it: 

```suggestion
Ory Oathkeeper is case-insensitive when accepting custom request headers. However, all 
incoming custom headers are converted to the canonical format of the MIME header key. 
This means that the first letter of the incoming header, as well as any letter that follows a hyphen, 
is converted into upper case and the rest of the letters are converted into lowercase.
For example, the incoming header `x-user-company` is converted and returned by
Oathkeeper as `X-User-Company`.
```

Does it sound okay? 

Not an easy thing to convey in writing :D

**Review by vinckr — APPROVED** [review]
LGTM 

Do you want to do a final review before we merge @tomekpapiernik ?

**Pull request merged**
